### PR TITLE
Updating Arango and Ent with KnownSince

### PR DESCRIPTION
**Pull request by nathannaveen**
# Description of the PR

* Updating `sbom`, `certifyGood` and `certifyBad` with `KnownSince`
* Fixes https://github.com/guacsec/guac/issues/1030.
* And is based on https://github.com/guacsec/guac/pull/1338#discussion_r1343080326.

# PR Checklist

- [ ] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

**Comment by nathannaveen** [comment]
@mrizzi @pxp928  👀

**Comment by pxp928** [comment]
@nathannaveen need a quick rebase here

File: pkg/assembler/backends/ent/backend/sbom.go [hunk 1]
```diff
@@ -80,13 +81,16 @@ func (b *EntBackend) IngestHasSbom(ctx context.Context, subject model.PackageOrA
 			SetDigest(strings.ToLower(spec.Digest)).
 			SetDownloadLocation(spec.DownloadLocation).
 			SetOrigin(spec.Origin).
-			SetCollector(spec.Collector)
+			SetCollector(spec.Collector).
+			SetKnownSince(spec.KnownSince)
```
Comment (pxp928) [inline pkg/assembler/backends/ent/backend/sbom.go:85]: this should be UTC like how you did the arango changes

File: pkg/assembler/backends/ent/backend/certify.go [hunk 2]
```diff
@@ -160,13 +161,15 @@ func upsertCertification[T certificationInputSpec](ctx context.Context, client *
...
+			SetCollector(v.Collector).
+			SetKnownSince(v.KnownSince)
 	case model.CertifyGoodInputSpec:
 		insert.
 			SetType(certification.TypeGOOD).
 			SetJustification(v.Justification).
 			SetOrigin(v.Origin).
-			SetCollector(v.Collector)
+			SetCollector(v.Collector).
+			SetKnownSince(v.KnownSince)
```
Comment (pxp928) [inline pkg/assembler/backends/ent/backend/certify.go:172]: UTC

File: pkg/assembler/backends/ent/backend/sbom.go (hunk 1 above)
Comment (nathannaveen) [reply pkg/assembler/backends/ent/backend/sbom.go]: Changed, thanks!

File: pkg/assembler/backends/ent/backend/certify.go (hunk 2 above)
Comment (nathannaveen) [reply pkg/assembler/backends/ent/backend/certify.go]: Changed

**Review by mrizzi — APPROVED** [review]
Fine on the Ent side as well, thanks @nathannaveen

**Pull request merged**